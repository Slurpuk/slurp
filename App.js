import 'react-native-gesture-handler';
import React, {useEffect, useState} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HamburgerSlideBarNavigator from './src/navigation/HamburgerSlideBarNavigator';
import SignUpPage from './src/screens/SignUpPage';
import LogInPage from './src/screens/LogInPage';
import WelcomePages from './src/screens/WelcomePages';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import AsyncStorage from '@react-native-async-storage/async-storage';
import firebase from '@react-native-firebase/app';
import firestore from '@react-native-firebase/firestore';
import {Alert} from 'react-native';
import auth from '@react-native-firebase/auth';

export const GlobalContext = React.createContext();
export default function App() {
  const [isFirstTime, setIsFirstTime] = useState(true);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [currentUser, setCurrentUser] = useState(auth().currentUser);
  const [userRef, setUserRef] = useState(null);
  const [userObj, setUserObj] = useState(null);
  const [shopsData, setShopsData] = useState([]);
  const [isShopIntro, setIsShopIntro] = useState(false);
  const [currShop, setCurrShop] = useState(shopsData[0]);
  const [isFullScreen, setFullScreen] = useState(false);
  const [basketContent, setBasketContent] = useState([]);
  const [basketSize, setBasketSize] = useState(0);
  const [total, setTotal] = useState(0);

  const checkForFirstTime = async () => {
    const result = await AsyncStorage.getItem('isFirstTime');
    //if what we get from the Async is null we are opening the app for the first time
    //if we pressed the sign up button on the last slide we set the 'isFirstTime' to 'no'
    if (result === null) {
      setIsFirstTime(true);
    } //now we can use the isFirstTimeLoad state to choose what to render
  };

  useEffect(() => {
    checkForFirstTime();
  }, []);

  useEffect(() => {
    const subscriber = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        setIsLoggedIn(true);
        setCurrentUser(user);
        setUser();
      } else {
        setIsLoggedIn(false);
        setCurrentUser(null);
      }
    });
    // Unsubscribe from events when no longer in use
    return () => subscriber();
  });

  const enterApp = () => {
    setIsFirstTime(false);
    AsyncStorage.setItem('isFirstTime', 'potatoesInPower');
  };

  function clearBasket() {
    setBasketContent([]);
    setBasketSize(0);
    setTotal(0);
  }

  function newShop({shop, navigation}) {
    clearBasket();
    setCurrShop(shop);
    navigation.navigate('Shop page');
  }

  useEffect(() => {
    const subscriber = firestore()
      .collection('Users')
      .doc(userRef)
      .onSnapshot(documentSnapshot => {
        setUserObj(documentSnapshot.data());
      });

    // Stop listening for updates when no longer required
    return () => subscriber();
  }, [userRef]);

  async function setUser() {
    if (currentUser) {
      await firestore()
        .collection('Users')
        .where('authID', '==', currentUser.uid)
        .get()
        .then(querySnapshot => {
          querySnapshot.forEach(documentSnapshot => {
            setUserRef(documentSnapshot.id);
          });
        });
    }
  }

  function changeShop({shop, navigation}) {
    if (currShop !== shop && basketSize !== 0) {
      Alert.alert(
        'Are you sure ?',
        'Changing shops will clear your basket.',
        [
          {
            text: 'Yes',
            onPress: () => newShop({shop, navigation}),
          },
          {
            text: 'No',
            onPress: () => navigation.navigate('Shop page'),
            style: 'cancel',
          },
        ],
        {cancelable: false},
      );
    } else {
      setCurrShop(shop);
      navigation.navigate('Shop page');
    }
  }

  // Subscribe to the Shops model
  useEffect(() => {
    const subscriber = firestore()
      .collection('CoffeeShop')
      .onSnapshot(querySnapshot => {
        const shops = [];

        querySnapshot.forEach(documentSnapshot => {
          let shopData = {
            ...documentSnapshot.data(),
            key: documentSnapshot.id,
          };

          let coffees = [];
          let drinks = [];
          let snacks = [];
          documentSnapshot.data().ItemsOffered.forEach(itemRef => {
            firestore()
              .doc(itemRef.path)
              .onSnapshot(querySnapshot => {
                let collection = '';
                if (itemRef.path.includes('Coffees')) {
                  collection = coffees;
                } else if (itemRef.path.includes('Drinks')) {
                  collection = drinks;
                } else {
                  collection = snacks;
                }
                collection.push({
                  ...querySnapshot.data(),
                  key: querySnapshot.id,
                });
              });
          });
          shopData.ItemsOffered = {
            Coffees: coffees,
            Drinks: drinks,
            Snacks: snacks,
          };
          shops.push(shopData);
          setShopsData(shops);
          setCurrShop(shops[0]);
        });
      });

    // Unsubscribe from events when no longer in use
    return () => subscriber();
  }, []);

  function isSameItem(it, currIt) {
    if (currIt.hasOwnProperty('Bean') && it.hasOwnProperty('Bean')) {
      let itOptions = '';
      it.options.forEach(option => (itOptions += option.Name));
      let currItOptions = '';
      currIt.options.forEach(option => (currItOptions += option.Name));
      return it.key === currIt.key && itOptions === currItOptions;
    } else {
      return it.key === currIt.key;
    }
  }

  function addToBasket(item) {
    const basket = basketContent;
    const exist = basket.find(x => isSameItem(x, item));
    if (exist) {
      setBasketContent(
        basket.map(x =>
          isSameItem(x, item) ? {...exist, count: exist.count + 1} : x,
        ),
      );
    } else {
      setBasketContent([...basket, {...item, count: 1}]);
    }
    setTotal(total + item.Price);
    setBasketSize(basketSize + 1);
  }

  function removeFromBasket(item) {
    const basket = basketContent;
    const exist = basket.find(x => isSameItem(x, item));
    if (exist.count === 1) {
      setBasketContent(basket.filter(x => x.key !== item.key));
    } else {
      setBasketContent(
        basket.map(x =>
          isSameItem(x, item) ? {...exist, count: exist.count - 1} : x,
        ),
      );
    }
    setTotal(total - item.Price);
    setBasketSize(basketSize - 1);
  }

  const setShopIntro = () => {
    setIsShopIntro(!isShopIntro);
  };

  const Stack = createNativeStackNavigator();
  return (
    <GlobalContext.Provider
      value={{
        enterApp: enterApp,
        user: currentUser, // Returns the authentication object
        currShop: currShop,
        setCurrShop: changeShop,
        isShopIntro: isShopIntro,
        shopsData: shopsData,
        setShopIntro: setShopIntro,
        isFullScreen: isFullScreen,
        setFullScreen: setFullScreen,
        basketContent: basketContent,
        setBasketContent: setBasketContent,
        total: total,
        setTotal: setTotal,
        addToBasket: addToBasket,
        removeFromBasket: removeFromBasket,
        basketSize: basketSize,
        clearBasket: clearBasket,
        currentUser: userObj, // Returns the model object
        userRef: userRef,
      }}
    >
      <NavigationContainer>
        {isLoggedIn ? (
          <HamburgerSlideBarNavigator />
        ) : (
          <Stack.Navigator
            screenOptions={{
              headerShown: false,
            }}
          >
            {isFirstTime ? (
              <Stack.Screen name="Welcome" component={WelcomePages} />
            ) : null}
            <Stack.Screen name="LogIn" component={LogInPage} />
            <Stack.Screen name="SignUp" component={SignUpPage} />
          </Stack.Navigator>
        )}
      </NavigationContainer>
    </GlobalContext.Provider>
  );
}
