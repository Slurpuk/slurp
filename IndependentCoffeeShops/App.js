import React, {useState} from 'react';
import {Pressable} from 'react-native';
import {BottomSheet} from 'react-native-bottomsheet-reanimated';
import type {Node} from 'react';
//import Inputs from 'DesignMap/src/screens/inputs.js'
import {
  View,
  Text,
  TouchableOpacity,
  TextInput,
  StyleSheet,
} from 'react-native';
import PrimaryButton from './PrimaryButton';
import textStyles from './stylesheets/textStyles';
import {authentication} from './firestore/firebase-config';
import firebase from '@react-native-firebase/app';
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';
import SignUpPage from "./components/SignUpPage";
import LoginPage from "./components/LoginPage";
import {rgbaColor} from 'react-native-reanimated/src/reanimated2/Colors';

export default function App() {
  const usersCollection = firestore().collection('Users');
  const [isSignedIn, setIsSignedIn] = useState(false);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [x, setX] = useState(usersCollection);

  const registerUser = () => {
    auth()
      .createUserWithEmailAndPassword(email, password)
      .then(re => {
        console.log(re);
        console.log(x);
      })
      .catch(re => {
        console.log(re);
      })
  }

  return (
    <View style={styles.mainContainer}>
      <TextInput
        placeholder="email"
        value={email}
        onChangeText={text => setEmail(text)}
      />
      <TextInput
        placeholder="password"
        value={password}
        secureTextEntry={true}
        onChangeText={text => setPassword(text)}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    marginTop: 55,
    flex: 1,
    padding: 10,
  },
});
