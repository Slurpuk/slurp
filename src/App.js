import 'react-native-gesture-handler';
import React, {useContext, useEffect, useReducer} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HamburgerSlideBarNavigator from './navigation/HamburgerSlideBarNavigator';
import SignUpPage from './screens/SignUpPage';
import LogInPage from './screens/LogInPage';
import WelcomePages from './screens/WelcomePages';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import firestore from '@react-native-firebase/firestore';
import auth from '@react-native-firebase/auth';
import LoadingPage from './screens/LoadingPage';
import {Alerts} from './data/Alerts';
import {
  formatCurrentOrders,
  formatOrders,
  formatPastOrders,
  getFormattedShops,
  loginUser,
  logoutUser,
  refreshOrders,
  refreshCurrentShop,
} from './helpers/screenHelpers';
import {getIsFirstTime, refreshCurrentBasket} from './helpers/storageHelpers';
import {firebase} from '@react-native-firebase/database';
import {GlobalAction} from './data/actionEnum';
import {GlobalContext} from './contexts';
import {globalReducer} from './reducers';
import {CurrentOrderStatuses, PastOrderStatuses} from './data/OrderStatus';

/**
 * Root component rendered when the application boots.
 */
export default function App() {
  const initialState = useContext(GlobalContext);
  const [globalState, globalDispatch] = useReducer(globalReducer, initialState);
  const LoggedOutStack = createNativeStackNavigator(); // Stack navigator for logged out users.
  const DB_URL =
    'https://slurp-59784-default-rtdb.europe-west1.firebasedatabase.app/';

  /**
   * Side effect that fires when App first renders to determine whether it is the first time the app is used since download
   * Sets the first time state accordingly.
   */
  useEffect(() => {
    async function setIsFirstTime() {
      const isFirstTime = await getIsFirstTime();
      if (!isFirstTime) {
        globalDispatch({type: GlobalAction.USE_APP});
      }
    }

    setIsFirstTime().catch(() => Alerts.elseAlert());
  }, []);

  /**
   * Side effect that tracks the authentication state of the current user.
   * Sets the current user object accordingly.
   */
  useEffect(() => {
    const subscriber = auth().onAuthStateChanged(async user => {
      if (user) {
        await loginUser(user, globalDispatch);
      } else {
        await logoutUser(globalDispatch);
      }
    });
    // Unsubscribe from events when no longer in use
    return () => subscriber();
  }, []);

  /**
   * Side effect that tracks changes in the model instance of the user in the database and updates the state accordingly
   * Sets the current user object accordingly.
   */
  useEffect(() => {
    if (!globalState.isLoadingUser) {
      const subscriber = firestore()
        .collection('users')
        .where('email', '==', auth().currentUser.email)
        .onSnapshot(query => {
          let newUserDoc = query.docs[0];
          let newUser = newUserDoc.data();
          globalDispatch({
            type: GlobalAction.SET_CURRENT_USER,
            user: {...newUser, key: newUserDoc.id, ref: newUserDoc.ref},
          });
        });
      return () => subscriber();
    }
  }, [globalState.isLoadingUser]);

  /**
   * Side effect that tracks changes in the current orders of the logged-in user.
   */
  useEffect(() => {
    if (!globalState.isLoadingUser) {
      const subscriber = firestore()
        .collection('orders')
        .where('is_displayed', '==', true)
        .where('user', '==', globalState.currentUserRef)
        .where('status', 'in', CurrentOrderStatuses)
        .onSnapshot(async query => {
          let currentOrders = await formatCurrentOrders(
            formatOrders(query.docs),
          );
          globalDispatch({
            type: GlobalAction.SET_CURRENT_ORDERS,
            orders: currentOrders,
          });
        });
      return () => subscriber();
    }
  }, [globalState.currentUserRef, globalState.isLoadingUser]);

  /**
   * Side effect that tracks changes in the past orders of the logged-in user.
   */
  useEffect(() => {
    if (!globalState.isLoadingUser) {
      const subscriber = firestore()
        .collection('orders')
        .where('is_displayed', '==', true)
        .where('user', '==', globalState.currentUserRef)
        .where('status', 'in', PastOrderStatuses)
        .onSnapshot(async query => {
          let pastOrders = await formatPastOrders(formatOrders(query.docs));
          globalDispatch({
            type: GlobalAction.SET_PAST_ORDERS,
            orders: pastOrders,
          });
        });
      return () => subscriber();
    }
  }, [globalState.currentUserRef, globalState.isLoadingUser]);

  /**
   * Side effect that tracks any change in the coffee shop model.
   * Retrieves the latest shop data, formats it and updates the current set of shops.
   */
  useEffect(() => {
    if (!globalState.isLoadingUser) {
      const subscriber = firestore()
        .collection('coffee_shops')
        .onSnapshot(async querySnapshot => {
          let formattedShops = await getFormattedShops(querySnapshot);
          globalDispatch({
            type: GlobalAction.SET_LIST_OF_SHOPS,
            shops: formattedShops,
          });
          await refreshCurrentShop(
            formattedShops,
            globalState.currentShopKey,
            globalDispatch,
          );
        });
      // Unsubscribe from events when no longer in use
      return () => subscriber();
    }
  }, [globalState.currentShopKey, globalState.isLoadingUser]);

  /**
   * Side effect that refreshes the logged-in user's data once the App loads.
   */
  useEffect(() => {
    async function refreshUserData() {
      if (!globalState.isLoadingUser) {
        await refreshCurrentBasket(globalDispatch);
        await refreshOrders(globalState.currentUserRef, globalDispatch);
      }
    }

    refreshUserData().then(() => console.log('refreshed'));
  }, [globalState.currentUserRef, globalState.isLoadingUser]);

  /**
   * Side effect that tracks the connection state of the app.
   */
  useEffect(() => {
    if (!globalState.isLoadingUser) {
      const uid = firebase.auth().currentUser.uid;

      const userStatusFirestoreRef = firestore()
        .collection('connection_status')
        .doc(uid);

      const userStatusDatabaseRef = firebase
        .app()
        .database(DB_URL)
        .ref('/status/' + uid);

      const isOfflineForDatabase = {
        status: 'offline',
        last_changed: firebase.database.ServerValue.TIMESTAMP,
      };
      const isOnlineForDatabase = {
        status: 'online',
        last_changed: firebase.database.ServerValue.TIMESTAMP,
      };
      const isOfflineForFirestore = {
        status: 'offline',
        last_changed: firebase.firestore.FieldValue.serverTimestamp(),
      };
      const isOnlineForFirestore = {
        status: 'online',
        last_changed: firebase.firestore.FieldValue.serverTimestamp(),
      };

      firebase
        .app()
        .database(DB_URL)
        .ref('.info/connected')
        .on('value', function (snapshot) {
          if (snapshot.val() === false) {
            userStatusFirestoreRef.set(isOfflineForFirestore);
            return;
          }
          userStatusDatabaseRef
            .onDisconnect()
            .set(isOfflineForDatabase)
            .then(async function () {
              userStatusDatabaseRef.set(isOnlineForDatabase);
              await userStatusFirestoreRef.set(isOnlineForFirestore);
            });
        });

      const subscriber = userStatusFirestoreRef.onSnapshot(function (doc) {
        const isOnline = doc.data().status === 'online';
        isOnline
          ? globalDispatch({type: GlobalAction.CONNECT})
          : globalDispatch({type: GlobalAction.DISCONNECT});
      });

      return () => subscriber();
    }
  }, [globalState.isLoadingUser]);

  return (
    <GlobalContext.Provider value={{globalState, globalDispatch}}>
      <NavigationContainer>
        {!globalState.isLoadingUser ? (
          <HamburgerSlideBarNavigator />
        ) : auth().currentUser ? (
          <LoadingPage />
        ) : (
          <LoggedOutStack.Navigator
            screenOptions={{
              headerShown: false,
            }}>
            {!globalState.isFirstTime ? (
              <LoggedOutStack.Screen name="Welcome" component={WelcomePages} />
            ) : null}
            <LoggedOutStack.Screen
              name="LogIn"
              children={props => <LogInPage {...props} />}
            />
            <LoggedOutStack.Screen
              name="SignUp"
              children={props => <SignUpPage {...props} />}
            />
          </LoggedOutStack.Navigator>
        )}
      </NavigationContainer>
    </GlobalContext.Provider>
  );
}
