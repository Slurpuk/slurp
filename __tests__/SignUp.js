import firestore from '@react-native-firebase/firestore';
import {render} from '@testing-library/react';
import App from "../App";

describe('test', function () {
  // beforeAll(async () => {
  //   await firestore()
  //     .collection('Users')
  //     .add({
  //       Name: 'Name',
  //       Email: 'me@name.com',
  //     })
  //     .then(() => console.log('Write'))
  //     .catch(error => {
  //       console.log(error);
  //     });
  // });
  it('test', function () {
    render(<App />);
    expect(true).toBe(true);
  });
});
