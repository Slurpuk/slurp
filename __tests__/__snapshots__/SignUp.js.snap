// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sing up should sign up a user 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EDEBE7",
      "display": "flex",
      "flex": 1,
      "paddingBottom": "5%",
      "paddingHorizontal": "5%",
      "paddingTop": undefined,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#173C4F",
          "fontFamily": "Josefin Sans",
          "fontSize": 35,
          "fontWeight": "700",
          "letterSpacing": 0.3,
        },
      ]
    }
  >
    Sign Up
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": "5%",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingVertical": "2%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginRight": "5%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#173C4F",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "includeFontPadding": false,
                "letterSpacing": 0.3,
              },
              Object {
                "marginBottom": "2.5%",
              },
            ]
          }
        >
          First Name
        </Text>
        <TextInput
          autoCapitalize="words"
          autoCompleteType="name"
          autoCorrect={false}
          keyboardType="default"
          maxLength={1235}
          onChangeText={[Function]}
          placeholder="Jane"
          placeholderTextColor="#D3D3D3"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#F9F9F9",
              "borderRadius": 5,
              "color": "black",
              "height": 37,
              "marginBottom": "3.5%",
              "paddingHorizontal": 10,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#173C4F",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "includeFontPadding": false,
                "letterSpacing": 0.3,
              },
              Object {
                "marginBottom": "2.5%",
              },
            ]
          }
        >
          Last Name
        </Text>
        <TextInput
          autoCapitalize="words"
          autoCompleteType="name"
          autoCorrect={false}
          keyboardType="default"
          maxLength={1235}
          onChangeText={[Function]}
          placeholder="Doe"
          placeholderTextColor="#D3D3D3"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#F9F9F9",
              "borderRadius": 5,
              "color": "black",
              "height": 37,
              "marginBottom": "3.5%",
              "paddingHorizontal": 10,
            }
          }
          value=""
        />
      </View>
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Email
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="email"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder="janedoe@gmail.com"
        placeholderTextColor="#D3D3D3"
        secureTextEntry={false}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Password
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="password"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder=""
        placeholderTextColor="#D3D3D3"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Confirm Password
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="password"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder=""
        placeholderTextColor="#D3D3D3"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#173C4F",
            "fontFamily": "Poppins-Regular",
            "fontSize": 15,
            "includeFontPadding": false,
            "letterSpacing": 0.3,
          },
          Object {
            "marginVertical": "2%",
            "textAlignVertical": "bottom",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Already have an account? Log in
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": "4%",
      }
    }
  >
    <View>
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#087562",
              "borderRadius": 13,
              "display": "flex",
              "flexDirection": "row",
              "height": 43,
              "justifyContent": "center",
              "opacity": 1,
              "width": 682.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EFEFEF",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 17,
                "letterSpacing": 0.5,
              }
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`sing up should sign up a user 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#EDEBE7",
      "display": "flex",
      "flex": 1,
      "paddingBottom": "5%",
      "paddingHorizontal": "5%",
      "paddingTop": undefined,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#173C4F",
          "fontFamily": "Josefin Sans",
          "fontSize": 35,
          "fontWeight": "700",
          "letterSpacing": 0.3,
        },
      ]
    }
  >
    Sign Up
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": "5%",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingVertical": "2%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginRight": "5%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#173C4F",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "includeFontPadding": false,
                "letterSpacing": 0.3,
              },
              Object {
                "marginBottom": "2.5%",
              },
            ]
          }
        >
          First Name
        </Text>
        <TextInput
          autoCapitalize="words"
          autoCompleteType="name"
          autoCorrect={false}
          keyboardType="default"
          maxLength={1235}
          onChangeText={[Function]}
          placeholder="Jane"
          placeholderTextColor="#D3D3D3"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#F9F9F9",
              "borderRadius": 5,
              "color": "black",
              "height": 37,
              "marginBottom": "3.5%",
              "paddingHorizontal": 10,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#173C4F",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "includeFontPadding": false,
                "letterSpacing": 0.3,
              },
              Object {
                "marginBottom": "2.5%",
              },
            ]
          }
        >
          Last Name
        </Text>
        <TextInput
          autoCapitalize="words"
          autoCompleteType="name"
          autoCorrect={false}
          keyboardType="default"
          maxLength={1235}
          onChangeText={[Function]}
          placeholder="Doe"
          placeholderTextColor="#D3D3D3"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#F9F9F9",
              "borderRadius": 5,
              "color": "black",
              "height": 37,
              "marginBottom": "3.5%",
              "paddingHorizontal": 10,
            }
          }
          value=""
        />
      </View>
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Email
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="email"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder="janedoe@gmail.com"
        placeholderTextColor="#D3D3D3"
        secureTextEntry={false}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Password
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="password"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder=""
        placeholderTextColor="#D3D3D3"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#173C4F",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "fontWeight": "600",
              "includeFontPadding": false,
              "letterSpacing": 0.3,
            },
            Object {
              "marginBottom": "2.5%",
            },
          ]
        }
      >
        Confirm Password
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCompleteType="password"
        autoCorrect={false}
        keyboardType="default"
        maxLength={1235}
        onChangeText={[Function]}
        placeholder=""
        placeholderTextColor="#D3D3D3"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "borderRadius": 5,
            "color": "black",
            "height": 37,
            "marginBottom": "3.5%",
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#173C4F",
            "fontFamily": "Poppins-Regular",
            "fontSize": 15,
            "includeFontPadding": false,
            "letterSpacing": 0.3,
          },
          Object {
            "marginVertical": "2%",
            "textAlignVertical": "bottom",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Already have an account? Log in
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": "4%",
      }
    }
  >
    <View>
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#087562",
              "borderRadius": 13,
              "display": "flex",
              "flexDirection": "row",
              "height": 43,
              "justifyContent": "center",
              "opacity": 1,
              "width": 682.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EFEFEF",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 17,
                "letterSpacing": 0.5,
              }
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
